### DISCUSSION TITLE
[RFC] Move cache outside of `node_modules`

### DISCUSSION THREAD

**Discussion by jaredpalmer** [vercel/turborepo#7497] [Ideas]
### Describe the feature you'd like to request

`node_modules` is a bad default place for the cache because most CI/CDs will blow out `node_modules` whenever the lockfile changes. The cache also currently grows in an unbounded manner. 

### Use `tmp` directory
On macOS and Linux consider using `tmp`.
 -  Pro: automatically clean up and not let the cache grow unbounded
 -  Con: hard to inspect from VSCode
 -  Con: hard to cache, will require custom flag to change position in CI so it is cacheable (--cache-dir=.turbo)

### Use `.turbo` at the root
- Pro: easy to inspect locally
- Pro: no needed to change location in ci w/flag

### Hybrid of both?
Use `tmp` on macOS when run in interactive shell, use .turbo in CI?

### Describe alternatives you've considered

n/a

**Comment by fz6m** [comment]
I think `${root}/.turbo` is better

1. **In github actions**: `.turbo` can be easily cached (see [`next.js`](https://github.com/vercel/next.js/blob/201f98e81a61a68d5d881a77d186308b0390ce92/.github/workflows/build_test_deploy.yml#L760))

2. **Local check**: When there is a problem with the cache, can delete it for check

**Comment by jlarmstrongiv** [comment]
Agree with @fz6m keep the turbo cache in the project folder please

**Comment by belgattitude** [comment]
Some thoughts, I'd prefer a .cache/turbo. But .turbo is fantastic

**Comment by weyert** [comment]
I would prefer the root of the project or an optional to configure where it should go

**Comment by miszo** [comment]
I would opt for the `Use .turbo at the root` option. It is easier to inspect locally and cache on CI.

**Comment by kevin-secrist** [comment]
Would be nice to specify the cache directory in turbo.json, in a similar way that yarn/npm allow you to change the package cache location in their config files. Seems like this would be a reasonable way to implement the feature without causing a breaking change. For caching in CI I've had to put `--cache-dir .turbo` in all calls to turbo, but it would be nice to avoid that by either specifying an env var or (preferably) putting it in the config file so it also applies to local development.

**Comment by anthonyshew** [comment]
Shipped in 2.0: https://github.com/vercel/turbo/pull/8126
